File: 3. Sophomore_HKI/OOP/OOP_CUOIKI/QuanLyShopQuanAo (C#)/ChiTietHoaDonNhapHang.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace QuanLyShopQuanAo
{
    public class ChiTietHoaDonNhapHang
    {
        public string maHoaDonNhapHang { get; set; }     // primary key
        public string maSanPham { get; set; }       // primary key
        public int soLuong { get; set; }
        public double donGia { get; set; }

        //singular
        public HoaDonNhapHang hoaDonNhapHang { get; set; }
        public SanPham sanPham { get; set; }

        // constructor
        public ChiTietHoaDonNhapHang() { }
        public ChiTietHoaDonNhapHang(ChiTietHoaDonNhapHang chiTietHoaDonNhap)
        {
            this.maHoaDonNhapHang = chiTietHoaDonNhap.maHoaDonNhapHang;
            this.maSanPham = chiTietHoaDonNhap.maSanPham;
            this.soLuong = chiTietHoaDonNhap.soLuong;
            this.donGia = chiTietHoaDonNhap.donGia;
            this.hoaDonNhapHang = chiTietHoaDonNhap.hoaDonNhapHang;
            this.sanPham = chiTietHoaDonNhap.sanPham;
        }
        public ChiTietHoaDonNhapHang(string maHoaDonNhapHang, string maSanPham, int soLuong, double donGia,
            HoaDonNhapHang hoaDonNhapHang, SanPham sanPham)
        {
            this.maHoaDonNhapHang = maHoaDonNhapHang;
            this.maSanPham = maSanPham;
            this.soLuong = soLuong;
            this.donGia = donGia;
            this.hoaDonNhapHang = hoaDonNhapHang;
            this.sanPham = sanPham;
        }
    }
}

File: 3. Sophomore_HKI/OOP/OOP_CUOIKI/QuanLyShopQuanAo (C#)/src/mypack/ChiTietHoaDonNhapHang.java
package mypack;

public class ChiTietHoaDonNhapHang {
	
	public String maHoaDonNhapHang; 	// primary key
    public String maSanPham; 			// primary key
    public int soLuong ;
    public double donGia ;

    //singular
    public HoaDonNhapHang hoaDonNhapHang;
    public SanPham sanPham;

	// constructor
    public ChiTietHoaDonNhapHang() { }
    public ChiTietHoaDonNhapHang(ChiTietHoaDonNhapHang cthd)
    {
        this.maHoaDonNhapHang = cthd.maHoaDonNhapHang;
        this.maSanPham = cthd.maSanPham;
        this.soLuong = cthd.soLuong;
        this.donGia = cthd.donGia;
        this.hoaDonNhapHang = cthd.hoaDonNhapHang;
        this.sanPham = cthd.sanPham;
    }
    public ChiTietHoaDonNhapHang(String maHoaDonNhapHang, String maSanPham, int soLuong, double donGia,
        HoaDonNhapHang hoaDonNhapHang, SanPham sanPham)
    {
        this.maHoaDonNhapHang = maHoaDonNhapHang;
        this.maSanPham = maSanPham;
        this.soLuong = soLuong;
        this.donGia = donGia;
        this.hoaDonNhapHang = hoaDonNhapHang;
        this.sanPham = sanPham;
    }   
    public ChiTietHoaDonNhapHang(String maSanPham, SanPham sanPham, double donGia, int soLuong)
    {
    	this.maSanPham = maSanPham;
        this.soLuong = soLuong;
        this.donGia = donGia;
        this.sanPham = sanPham;
    }
    
    // getter - setter
    public String getmaHoaDonNhapHang() {
		return maHoaDonNhapHang;
	}
	public void setmaHoaDonNhapHang(String maHoaDonNhapHang) {
		this.maHoaDonNhapHang = maHoaDonNhapHang;
	}
	public String getMaSanPham() {
		return maSanPham;
	}
	public void setMaSanPham(String maSanPham) {
		this.maSanPham = maSanPham;
	}
	public int getSoLuong() {
		return soLuong;
	}
	public void setSoLuong(int soLuong) {
		this.soLuong = soLuong;
	}
	public double getDonGia() {
		return donGia;
	}
	public void setDonGia(double donGia) {
		this.donGia = donGia;
	}
	public HoaDonNhapHang gethoaDonNhapHang() {
		return hoaDonNhapHang;
	}
	public void setHoaDonNhapHang(HoaDonNhapHang hoaDonNhapHang) {
		this.hoaDonNhapHang = hoaDonNhapHang;
	}
	public SanPham getSanPham() {
		return sanPham;
	}
	public void setSanPham(SanPham sanPham) {
		this.sanPham = sanPham;
	}   
}
